src/webparts/todolist/components/ToDoList.tsx

```diff
@@ -1,129 +1,213 @@
+            return (
+              <Stack
+                horizontalAlign="center"
+                verticalAlign="center"
+                className={mergeStyles({
+                  backgroundColor: "#ffe092",
+                  color: "#997825",
+                  borderRadius: "16px",
+                  height: "24px",
+                  whiteSpace: "nowrap",
+                  padding: "4px 8px",
+                  maxWidth: "max-content"
+                })}
+              >
+                {item.Status}
+              </Stack>
+            );
+
+          case "Completed":
+            return (
+              <Stack
+                horizontalAlign="center"
+                verticalAlign="center"
+                className={mergeStyles({
+                  backgroundColor: "#5dd4c0",
+                  color: "#006b59",
+                  borderRadius: "16px",
+                  height: "24px",
+                  whiteSpace: "nowrap",
+                  padding: "4px 8px",
+                  maxWidth: "max-content"
+                })}
+              >
+                {item.Status}
+              </Stack>
+            );
+
+          case "Active":
+            return (
+              <Stack
+                horizontalAlign="center"
+                verticalAlign="center"
+                className={mergeStyles({
+                  backgroundColor: "#8ac2ec",
+                  color: "#235a85",
+                  borderRadius: "16px",
+                  height: "24px",
+                  whiteSpace: "nowrap",
+                  padding: "4px 8px",
+                  maxWidth: "max-content"
+                })}
+              >
+                {item.Status}
+              </Stack>
+            );
+
+          case "Overdue":
+            return (
+              <Stack
+                horizontalAlign="center"
+                verticalAlign="center"
+                className={mergeStyles({
+                  backgroundColor: "#f6a89a",
+                  color: "#903e2f",
+                  borderRadius: "16px",
+                  height: "24px",
+                  whiteSpace: "nowrap",
+                  padding: "4px 8px",
+                  maxWidth: "max-content"
+                })}
+              >
+                {item.Status}
+              </Stack>
+            );
+
+          default:
+            return <Stack>{item.Status}</Stack>;
+        }
       }
-
-      const data = await response.json();
-
-      this.setState({ items: data.value });
-      return data.value;
-    } catch (error) {
-      this.setState({ errorMessage: error.message });
+    },
+    {
+      key: "column4",
+      name: "Description",
+      fieldName: "Description",
+      isMultiline: true,
+      minWidth: 100,
+      maxWidth: 200,
+      isResizable: true
     }
-  }
+  ];
 
-  public componentDidMount(): void {
-    this._getListData();
-  }
+  const getListData = async (): Promise<void> => {
+    const LIST_NAME = "To do list";
```

Comment:
List name should't be hardcoded, put it as an parameter to webpart properties and add ability to change it via property pane